src/core/tests/eval.cpp

```diff
@@ -1028,6 +1028,21 @@ TEST(eval, evaluate_sign) {
     ASSERT_EQ(result_val, expec);
 }
 
+TEST(eval, evaluate_sign_2) {
```

Comment:
Give better name to this test case to  indicate NaN values are tested.
I think there should be template plugin test for this operator and reference implementation. It would be better to update these test as they usually  perform calculations on different precisions.

Reply:
test case name changed